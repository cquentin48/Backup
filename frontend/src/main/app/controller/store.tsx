import { configureStore } from "@reduxjs/toolkit";

import deviceReducer from "./deviceMainInfos/loadDeviceSlice";
import filterReducer from "./deviceMainInfos/filterSlice";
import snapshotReducer from "./deviceMainInfos/loadSnapshotSlice";

export const store = configureStore({
    reducer: {
        device: deviceReducer,
        filter: filterReducer,
        snapshot: snapshotReducer
    }
})

export type AppState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
