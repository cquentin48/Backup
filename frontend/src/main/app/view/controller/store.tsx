import { configureStore } from "@reduxjs/toolkit";

import loadSnapshotReducer from "./deviceMainInfos/loadSnapshotSlice";
import loadDeviceReducer from "./deviceMainInfos/loadDeviceSlice";
import filterReducer from "./deviceMainInfos/filterSlice";

export const store = configureStore({
    reducer: {
        loadSnapshot: loadSnapshotReducer,
        filters: filterReducer,
        device: loadDeviceReducer
    }
})

export type AppState = ReturnType<typeof store.getState>;