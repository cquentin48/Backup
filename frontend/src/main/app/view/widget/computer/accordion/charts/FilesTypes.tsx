import React from "react";

import { mdiFileOutline } from "@mdi/js";
import Icon from "@mdi/react";

import { Avatar, Card, CardContent, CardHeader } from "@mui/material";
import { PieChart, ResponsiveChartContainer } from "@mui/x-charts";

import '../../../../../../res/css/ComputerMainInfos.css';

/**
 *  Files types pie chart view component
 *  @returns {React.JSX.Element} View component
 */
export default function FilesTypes (): React.JSX.Element {
    return (
        <Card className="PieChartCard">
            <CardHeader
                avatar={
                    <Avatar
                        sx={{ bgcolor: "#c2c2c2" }}
                        arial-labels="recipe">
                        <Icon path={mdiFileOutline} size={1} />
                    </Avatar>
                }
                title="Types of files"
            />
            <CardContent>
                    <PieChart
                        series={[
                            {
                                data: [
                                    { id: 0, value: 26, label: "Text/Other" },
                                    { id: 1, value: 50, label: "Images" },
                                    { id: 2, value: 75, label: "Videos" },
                                    { id: 3, value: 155, label: "Shared Libraries" },
                                    { id: 4, value: 75, label: "Music" },
                                    { id: 5, value: 155, label: "Other" }
                                ]
                            }
                        ]}
                        width={550}
                        height={200}
                        className="DisplayedPieChart"
                    />
            </CardContent>
        </Card>
    )
}
