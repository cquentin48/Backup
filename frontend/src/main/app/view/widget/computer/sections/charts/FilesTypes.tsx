import React from "react";

import { mdiFileOutline } from "@mdi/js";
import Icon from "@mdi/react";

import { Avatar, Card, CardContent, CardHeader } from "@mui/material";
import { PieChart, type PieSeriesType, type PieValueType } from "@mui/x-charts";

import '../../../../../../res/css/ComputerMainInfos.css';
import { type MakeOptional } from "@mui/x-charts/internals";
import Device from "../../../../../model/device/device";

/**
 * Types of files passed from the chart
 */
interface FilesTypesProps {
    /**
     * If the device has been loaded
     */
    device: Device
}

/**
 *  Files types pie chart view component
 *  @returns {React.JSX.Element} View component
 *  @param {FilesTypesProps} props If the device has been loaded function
 */
export default function FilesTypes (props: FilesTypesProps): React.JSX.Element {
    const series: Array<MakeOptional<PieSeriesType<MakeOptional<PieValueType, "id">>, "type">> = [
        {
            data: [
                { id: 0, value: 26, label: "Text/Other" },
                { id: 1, value: 50, label: "Images" },
                { id: 2, value: 75, label: "Videos" },
                { id: 3, value: 155, label: "Shared Libraries" },
                { id: 4, value: 75, label: "Music" },
                { id: 5, value: 155, label: "Other" }
            ]
        }
    ]
    return (
        <Card className="PieChartCard">
            <CardHeader
                avatar={
                    <Avatar
                        className="AvatarIcon"
                        arial-labels="recipe">
                        <Icon path={mdiFileOutline} size={1} />
                    </Avatar>
                }
                title="Types of files"
            />
            <CardContent>
                <PieChart
                    loading={props.device === undefined}
                    series={series}
                    width={550}
                    height={200}
                    className="DisplayedPieChart"
                />
            </CardContent>
        </Card>
    )
}
