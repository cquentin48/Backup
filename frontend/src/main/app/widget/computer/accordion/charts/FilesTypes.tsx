import { mdiFileOutline, mdiFilterOutline, mdiPlus } from "@mdi/js";
import Icon from "@mdi/react";

import { Avatar, Card, CardContent, CardHeader, IconButton, Menu, MenuItem, Switch, TextField } from "@mui/material";
import { PieChart } from "@mui/x-charts";
import React from "react";

export default function FilesTypes() {

    const [anchorEl, setanchorEl] = React.useState<null | HTMLElement>(null);
    const open = Boolean(anchorEl);
    const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
        setanchorEl(event.currentTarget);
    }
    const handleClose = (event: React.MouseEvent<HTMLButtonElement>) => {
        setanchorEl(null);
    }

    return (
        <Card sx={{ width: "fit-content" }}>
            <CardHeader
                avatar={
                    <Avatar
                        sx={{ bgcolor: "#c2c2c2" }}
                        arial-labels="recipe">
                        <Icon path={mdiFileOutline} size={1} />
                    </Avatar>
                }
                title="Types of files"
                action={
                    <IconButton
                        aria-label="settings"
                        aria-controls={open ? 'basic-menu' : undefined}
                        aria-haspopup="true"
                        aria-expanded={open ? 'true' : undefined}
                        onClick={handleClick}>
                        <Icon path={mdiFilterOutline} size={1} />
                        <Menu
                            id="file-types-filter-menu"
                            anchorEl={anchorEl}
                            open={open}
                            onClose={handleClose}
                            MenuListProps={{
                                'aria-labelledby': 'basic-button'
                            }}
                        >
                            <MenuItem>
                                No filter added.
                                <IconButton>
                                    <Icon path={mdiPlus} size={1}/>
                                </IconButton>
                            </MenuItem>
                        </Menu>
                    </IconButton>
                }
            />
            <CardContent>
                <PieChart
                    series={[
                        {
                            data: [
                                { id: 0, value: 26, label: "Text/Other" },
                                { id: 1, value: 50, label: "Images" },
                                { id: 2, value: 75, label: "Videos" },
                                { id: 3, value: 155, label: "Shared Libraries" },
                                { id: 4, value: 75, label: "Music" },
                                { id: 5, value: 155, label: "Other" }
                            ]
                        }
                    ]}
                    width={550}
                    height={200}
                    className="DisplayedPieChart"
                />
            </CardContent>
        </Card>
    )
}