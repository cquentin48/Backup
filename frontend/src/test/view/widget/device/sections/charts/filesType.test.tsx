import React from "react"

import { render } from "@testing-library/react"
import FilesTypes from "../../../../../../main/app/view/pages/computer/sections/charts/FilesTypes"

import '@testing-library/jest-dom'
import { Provider } from "react-redux"
import store from "../../../../../../main/app/controller/store"

describe("Type of files chart unit test suite", () => {
    test("Successful render", async () => {
        // Acts
        const { asFragment } = render(
            <Provider store={store}>
                <FilesTypes />
            </Provider>
        )

        // Acts
        expect(asFragment()).toMatchSnapshot
    })
})
