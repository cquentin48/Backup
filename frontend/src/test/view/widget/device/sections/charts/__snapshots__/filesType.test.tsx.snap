// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Type of files chart unit test suite Successful render 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root PieChartCard css-1j74uew-MuiPaper-root-MuiCard-root"
    style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiCardHeader-root css-149tc0y-MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-avatar css-1r9wl67-MuiCardHeader-avatar"
      >
        <div
          arial-labels="recipe"
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault AvatarIcon css-18k2bs-MuiAvatar-root"
        >
          <svg
            role="presentation"
            style="width: 1.5rem; height: 1.5rem;"
            viewBox="0 0 24 24"
          >
            <path
              d="M14,2H6A2,2 0 0,0 4,4V20A2,2 0 0,0 6,22H18A2,2 0 0,0 20,20V8L14,2M18,20H6V4H13V9H18V20Z"
              style="fill: currentColor;"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiCardHeader-content css-1jkp9bu-MuiCardHeader-content"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-16xl4zq-MuiTypography-root"
        >
          Types of files
        </h5>
      </div>
    </div>
    <div
      class="MuiCardContent-root css-1lt5qva-MuiCardContent-root"
    >
      <div
        class="DisplayedPieChart css-ba51a4-MuiResponsiveChart-container"
      >
        <svg
          class="css-1evyvmv-MuiChartsSurface-root"
          height="200"
          viewBox="0 0 550 200"
          width="550"
        >
          <title />
          <desc />
          <defs />
          <g>
            <g
              transform="translate(227.5, 100)"
            />
            <g
              transform="translate(227.5, 100)"
            />
          </g>
          <text
            class="css-1f57y8b"
            x="227.5"
            y="100"
          >
            Loading dataâ€¦
          </text>
          <g
            class="MuiChartsLegend-root MuiChartsLegend-column css-nsbw82-MuiChartsLegend-root"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
